kernel.rb

```diff
@@ -169,6 +169,10 @@ def yield_self
   #     Float("123.0_badstring", exception: false)  #=> nil
   #
   def Float(arg, exception: true)
-    Primitive.rb_f_float(arg, exception)
+    if Primitive.mandatory_only?
+      Primitive.rb_f_float1(arg)
```

Comment:
Does this also achieve similar improvement? If it does, it might be better for maintainability to not introduce a new C function.
```suggestion
      Primitive.rb_f_float(arg, true)
```

Reply:
@k0kubun 
OK, I'll try it.

Reply:
It seems like I was misunderstanding the internal a bit. IIUC, it has to be this form to generate `opt_invokebuiltin_delegate_leave` insn and my suggestion would change it to `invokebuiltin`, which is slower. So we probably have to keep this. It'd be interesting to see the benchmark difference though.